File: digitalpassbook2/app/src/main/java/com/example/digitalpassbook/EventService.java
package com.example.digitalpassbook;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.PATCH;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface EventService {
    @GET("events")
    Call<List<Event>> getall();

    @POST("events/new")
    Call<Event> create(@Body Event event);

    @GET("events/{id}")
    Call<Event> get(@Path("id") int id);

    @DELETE("events/{id}")
    Call<Event> delete(@Path("id") int id);

    @PATCH("events/{id}")
    Call<Event> update(@Path("id") int id, @Body Event event);
}

File: digitalpassbook2/app/src/main/java/com/example/digitalpassbook2/EventService.kt
package com.example.digitalpassbook2

import retrofit2.Call
import retrofit2.http.*

interface EventService {
    @GET("events")
    fun getall(): Call<List<Event?>?>?

    @POST("events/new")
    fun create(@Body event: Event?): Call<Event?>?

    @GET("events/{id}")
    operator fun get(@Path("id") id: Int): Call<Event?>?

    @DELETE("events/{id}")
    fun delete(@Path("id") id: Int): Call<Event?>?

    @PATCH("events/{id}")
    fun update(
        @Path("id") id: Int,
        @Body event: Event?
    ): Call<Event?>?
}